spark-etl/src/main/scala/geotrellis/spark/etl/Etl.scala

```diff
@@ -72,35 +75,75 @@ case class Etl(args: Seq[String], @transient modules: Seq[TypedModule] = Etl.def
     plugin(conf.inputProps)
   }
 
-  def reproject[
-  K: SpatialComponent: ClassTag,
-  V <: CellGrid: ClassTag: Stitcher: ? => TileReprojectMethods[V]: ? => CropMethods[V]: ? => TileMergeMethods[V]: ? => TilePrototypeMethods[V]
-  ](rdd: RDD[(K, V)] with Metadata[M]): (Int, RDD[(K, V)] with Metadata[M]) = {
-    val tuple = rdd.reproject(conf.crs(), conf.layoutScheme()(conf.crs(), conf.tileSize()))
-    tuple._2.persist(conf.cache())
-    tuple
-  }
+  /**
+    * Tiles RDD of arbitrary rasters to conform to a layout scheme or definition provided in the arguments.
+    * First metadata will be collected over input rasters to determine the overall extent, common crs, and resolution.
+    * This information will be used to select a LayoutDefinition if LayoutScheme is provided in the arguments.
+    *
+    * The tiling step will use this LayoutDefinition to cut input rasters into chunks that conform to the layout.
+    * If multiple rasters contribute to single target tile their values will be merged cell by cell.
+    *
+    * After the tiling step a buffered reproject will be performed.
+    * The buffered reproject will perform neighborhood operation, where cells from surrounding tiles will be sampled.
+    *
+    * @param rdd    RDD of source rasters
+    * @param method Resampling method to be used when merging raster chunks in tiling step
+    */
+  def tile[
+    I: ProjectedExtentComponent: ? => TilerKeyMethods[I, K],
+    V <: CellGrid: Stitcher: ClassTag: (? => TileMergeMethods[V]): (? => TilePrototypeMethods[V]):
+      (? => TileReprojectMethods[V]): (? => CropMethods[V]),
+    K: SpatialComponent: Boundable: ClassTag
+  ](
+    rdd: RDD[(I, V)], method: ResampleMethod = NearestNeighbor
+  )(implicit sc: SparkContext): (Int, RDD[(K, V)] with Metadata[RasterMetaData[K]]) = {
+    val targetCellType = conf.cellType.get
+
+    val (_, rmd: RasterMetaData[K]) = {
```

Comment:
Should add an option so that we can reproject the source tiles via ProjectedComponentMethods reproject vs the buffered tile reprojection method. The latter requires that all the incoming CRS's are the same; we could handle the case where a buffered reproject is requested but not all the CRS's are the same inside the metadata collect, and report that error.